src/background/autoconsent.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+import AutoConsent from '@cliqz/autoconsent';
+
+const config = { autoOptOut: true };
+
+const consent = new AutoConsent(chrome, (tabId, message, options) => {
+  return new Promise((resolve) => {
+    chrome.tabs.sendMessage(tabId, message, options, resolve);
+  });
+});
+
+(async function loadRules() {
+  const res = await fetch(
+    chrome.runtime.getURL('node_modules/@cliqz/autoconsent/rules/rules.json'),
+  );
+  const rules = await res.json();
+  Object.keys(rules.consentomatic).forEach((name) => {
+    consent.addConsentomaticCMP(name, rules.consentomatic[name]);
+  });
+  rules.autoconsent.forEach((rule) => {
+    consent.addCMP(rule);
+  });
+})();
+
+chrome.webNavigation.onCommitted.addListener(
```

Comment:
wonder if in context of Service Worker this is consider top level registration. we should probably move the listeners to the main file